hadoop-hdds/interface-server/src/main/proto/ScmServerProtocol.proto

```diff
@@ -132,9 +132,13 @@ message AllocateScmBlockRequestProto {
   required uint64 size = 1;
   required uint32 numBlocks = 2;
   required ReplicationType type = 3;
-  required hadoop.hdds.ReplicationFactor factor = 4;
+  optional hadoop.hdds.ReplicationFactor factor = 4;
```

Comment:
Is moving this field from required to optional OK from a forward / backwards compatibility point of view? Would an old client which has it set to required, complain if it received a message missing the field?

Old Client -> New Server:
Will only ever send the factor field | will be OK

New Server -> Old Client
Client can never understand EC. Other files would have factor set.

New Client -> Old Server
Old server cannot understand EC, must always send Factor for Ratis files.

Old Server -> New Client:
Will always sent factor

I think it will be OK, as there is a natural incompatibility between old client / server and new client / server which supports EC.

Reply:
> Would an old client which has it set to required, complain if it received a message missing the field?

Good question and I agree that it should be OK. This field will always be filled for STANDALONE / RATIS. Old clients remain compatible as long as they use only existing replication types. It is optional only for EC where we need an exception from the old client anyway, as old client couldn't use new replication types.